src/Compilers/CSharp/Portable/CodeGen/EmitArrayInitializer.cs

```diff
@@ -372,37 +372,61 @@ private static bool IsMultidimensionalInitializer(ImmutableArray<BoundExpression
 
 #nullable enable
 
-        private bool TryEmitReadonlySpanAsBlobWrapper(NamedTypeSymbol spanType, BoundExpression wrappedExpression, bool used, bool inPlace, BoundExpression? start = null, BoundExpression? length = null)
+        private bool TryEmitReadonlySpanAsBlobWrapper(NamedTypeSymbol spanType, BoundExpression wrappedExpression, bool used, BoundExpression inPlaceTarget, out bool avoidInPlace, BoundExpression? start = null, BoundExpression? length = null)
         {
             Debug.Assert(start is null == length is null);
 
             ImmutableArray<byte> data = default;
             int elementCount = -1;
+            MethodSymbol? createSpan = null;
+            avoidInPlace = false;
 
             if (!_module.SupportsPrivateImplClass)
             {
                 return false;
             }
 
             var ctor = ((MethodSymbol?)this._module.Compilation.GetWellKnownTypeMember(WellKnownMember.System_ReadOnlySpan_T__ctor_Pointer));
-            if (ctor == null)
+            if (ctor == null || ctor.GetUseSiteInfo().DiagnosticInfo?.Severity == DiagnosticSeverity.Error)
```

Comment:
>|| ctor.GetUseSiteInfo().DiagnosticInfo?.Severity == DiagnosticSeverity.Error [](http://example.com/codeflow?start=29&length=77)

Instead of performing a manual check like this, let's use an existing helper to get the symbol, which is also going to take care of proper dependency tracking:
```
var ctor = (MethodSymbol?)GetWellKnownTypeMember(this._module.Compilation, WellKnownMember.System_ReadOnlySpan_T__ctor_Pointer, _diagnostics, syntax: wrappedExpression.Syntaxl, isOptional: true);
if (ctor == null)
{
    return false;
}
```